### PR TITLE
add feat: subscription.py optional param for sampling_interval

### PR DESCRIPTION
**Pull request by AndreasHeine**

### CODE REVIEW

**Comment by AndreasHeine** [comment]
@oroulet tested against a "Unified Automation Cpp Server"!

File: asyncua/common/subscription.py [hunk 1]
```diff
@@ -171,12 +177,14 @@ async def _call_status(self, status: ua.StatusChangeNotification):
...
-                                    monitoring=ua.MonitoringMode.Reporting,
-                                   ) -> Union[int, List[Union[int, ua.StatusCode]]]:
+    async def subscribe_data_change(
+        self,
+        nodes: Union[Node, Iterable[Node]],
+        attr: ua.AttributeIds = ua.AttributeIds.Value,
+        queuesize: int = 0,
+        monitoring=ua.MonitoringMode.Reporting,
+        sampling_interval: ua.Duration = 0.0
+    ) -> Union[int, List[Union[int, ua.StatusCode]]]:
```
Comment (oroulet) [inline asyncua/common/subscription.py:187]: ideally we should pass nodes AND the entire MonitoredItemCreateRequest so people can add whatever they want. but that breaks the API....

**Comment by oroulet** [comment]
your PR is so hard to read because you have mixed up many formatting changes and typing with real code changes...

File: asyncua/common/subscription.py (hunk 1 above)
Comment (oroulet) [reply asyncua/common/subscription.py:187]: but create_monitored_items is hard to use, it does not any handler or anything, so you need to catch events by hand (I do not even remember how tot do that)
Anyway. let's do it the way you propose for now

**Pull request merged**